src/bridge/bridge.c

```diff
@@ -85,10 +88,28 @@ process_init (CockpitTransport *transport,
   else
     {
       g_message ("unsupported version of cockpit protocol: %" G_GINT64_FORMAT, version);
-      cockpit_transport_close (transport, "protocol-error");
+      cockpit_transport_close (transport, "not-supported");
     }
 }
 
+static struct {
+  const gchar *name;
+  GType (* function) (void);
+} payload_types[] = {
+  { "dbus-json1", cockpit_dbus_json1_get_type },
+  { "dbus-json3", cockpit_dbus_json_get_type },
+  { "http-stream1", cockpit_http_stream_get_type },
+  { "stream", cockpit_stream_get_type },
+  { "fsread1", cockpit_fsread_get_type },
+  { "fsreplace1", cockpit_fsreplace_get_type },
+  { "fswatch1", cockpit_fswatch_get_type },
+  { "fslist1", cockpit_fslist_get_type },
+  { "null", cockpit_null_channel_get_type },
+  { "echo", cockpit_echo_channel_get_type },
+  { "metrics1", cockpit_internal_metrics_get_type },
+  { NULL },
```

Comment:
if we're adding these now... should we sort them alphanumerically? or is there a specific order?
if the list is sorted, technically we could find a match a lot faster than we currently do - binary search instead of linear search.